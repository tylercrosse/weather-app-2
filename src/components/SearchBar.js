import React from 'react';
import PropTypes from 'prop-types';
import PlacesAutoComplete from 'react-places-autocomplete';

class SearchBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      address: ''
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSelect = this.handleSelect.bind(this);
  }
  handleChange(address) {
    this.setState({
      address
    });
  };
  handleSelect(address) {
    this.props.geocode(address);
    this.setState({
      address: ''
    });
  };
  render() {
    const AutocompleteItem = ({ formattedSuggestion }) =>
      <div className="search__item">
        <strong>{formattedSuggestion.mainText}</strong>{' '}
        <small className="text-muted">
          {formattedSuggestion.secondaryText}
        </small>
      </div>;
    const cssClasses = {
      root: 'search__bar',
      input: 'search__input',
      autocompleteContainer: 'search__autocomplete_container'
    };
    const inputProps = {
      type: 'text',
      value: this.state.address,
      onChange: this.handleChange,
      autoFocus: true,
      placeholder: 'Search Places'
    };

    return (
      <div className="search">
        <PlacesAutoComplete
          inputProps={inputProps}
          onSelect={this.handleSelect}
          onEnterKeyDown={this.handleSelect}
          classNames={cssClasses}
          autocompleteItem={AutocompleteItem}
          highlightFirstSuggestion
        />
      </div>
    );
  }
}

SearchBar.propTypes = {
  geocode: PropTypes.func.isRequired,
}

export default SearchBar;
