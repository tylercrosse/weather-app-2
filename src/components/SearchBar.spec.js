import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import SearchBar from './SearchBar';

const setup = () => {
  const props = {
    geocode: jest.fn(),
  }

  const component = <SearchBar {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<SearchBar />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
