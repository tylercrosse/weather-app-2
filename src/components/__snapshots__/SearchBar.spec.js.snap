// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render correctly 1`] = `
<div
  className="search"
>
  <PlacesAutocomplete
    autocompleteItem={[Function]}
    classNames={
      Object {
        "autocompleteContainer": "search__autocomplete_container",
        "input": "search__input",
        "root": "search__bar",
      }
    }
    clearItemsOnError={false}
    debounce={200}
    highlightFirstSuggestion={true}
    inputProps={
      Object {
        "autoFocus": true,
        "onChange": [Function],
        "placeholder": "Search Places",
        "type": "text",
        "value": "",
      }
    }
    onEnterKeyDown={[Function]}
    onError={[Function]}
    onSelect={[Function]}
    options={Object {}}
    styles={Object {}}
  />
</div>
`;
